lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -141,13 +141,20 @@ def inline_metadata(catalog, checksum_type)
 
           # TODO: Conditionally copy owner, group, and mode when we can check if permissions are set
           child_resource = Puppet::Resource.new(:file, File.join(file[:path], meta.relative_path))
+          child_resource[:ensure] = meta.ftype
           replace_metadata(child_resource, meta, true)
 
           # Copy parameters from original parent directory
           file.original_parameters.each do |param, value|
             # These should never be passed to our children
             unless [:parent, :ensure, :recurse, :recurselimit, :target, :alias, :source].include? param
-              child_resource[param] = value.to_s
+              # This logic belongs in the file type/individual parameters,
+              # e.g. param.copy(child_resource)
+              if [:before, :require, :subscribe, :notify].include? param
+                child_resource[param] = value
+              else
+                child_resource[param] = value.to_s
```

Comment:
Actually... I was using `to_s` because I was originally trying to use `file.parameters` rather than `file.original_parameters`. However, it may be the case that because `original_parameters` is just a plain old hash that it will return a string for the value and we may not need to call `to_s`. I haven't actually tested it either way though.